app/components/sidebars/drawer_layout.js

```diff
@@ -215,30 +219,18 @@ export default class DrawerLayout extends Component {
             transform: [{ translateX: drawerTranslateX }],
         };
 
-        /* Overlay styles */
-        const overlayOpacity = this.openValue.interpolate({
-            inputRange: [0, 1],
-            outputRange: [0, 0.5],
-            extrapolate: 'clamp',
-        });
-        const animatedOverlayStyles = { opacity: overlayOpacity };
-        const pointerEvents = drawerShown ? 'auto' : 'none';
+        // 0 - ios main drawer
+        // 1 - main | tablet
+        // 2 - overlay
+        // 3 - android main drawer | settings drawer
+        const drawerZIndex = isMain && Platform.OS === 'ios' ? 0 : 3
 
         return (
             <React.Fragment>
-                <TouchableWithoutFeedback
-                    pointerEvents={pointerEvents}
-                    onPress={this._onOverlayClick}
-                >
-                    <Animated.View
-                        pointerEvents={pointerEvents}
-                        style={[styles.overlay, animatedOverlayStyles]}
-                    />
-                </TouchableWithoutFeedback>
                 <Animated.View
                     accessibilityViewIsModal={accessibilityViewIsModal}
                     style={[
-                        styles.drawer,
+                        StyleFactory.drawer(drawerZIndex),
```

Comment:
interesting approach 👍

Reply:
This is awesome, I haven't tried this before